### DISCUSSION TITLE
How to configure a proxy for MTProxy

### DISCUSSION THREAD

**Discussion by VladislavGatsenko** [9seconds/mtg#293] [Q&A]
Hi! 

Does anyone know how to use NGINX to configure MTProxy access? 

I deployed the https://github.com/missuo/MTProxy server (with this script: bash <(curl -Ls https://cpp.li/mtg) ) 

It works on the standard port 8443. 

How do I connect to mydomain:433 without changing the MTProxy server configuration?

**Comment by Ashikurrahaman287** [comment]
To configure NGINX to proxy requests to your MTProxy server without changing the MTProxy server configuration, you can follow these steps:

1. Install NGINX (if not already installed) and start the service:

   ```bash
   sudo apt update
   sudo apt install nginx
   sudo systemctl start nginx
   ```

2. Create a new NGINX configuration file for your MTProxy server:

   ```bash
   sudo nano /etc/nginx/sites-available/mtproxy
   ```

3. Add the following configuration to the file, replacing `yourdomain.com` with your actual domain name:

   ```nginx
   server {
       listen 433 ssl;
       server_name yourdomain.com;

       ssl_certificate /etc/ssl/certs/yourdomain.com.crt;
       ssl_certificate_key /etc/ssl/private/yourdomain.com.key;

       location / {
           proxy_pass http://127.0.0.1:8443;
           proxy_set_header Host $host;
           proxy_set_header X-Real-IP $remote_addr;
           proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
           proxy_set_header X-Forwarded-Proto $scheme;
       }
   }
   ```

   Make sure to replace `/etc/ssl/certs/yourdomain.com.crt` and `/etc/ssl/private/yourdomain.com.key` with the paths to your SSL certificate and private key files.

4. Save the file and exit the text editor.

5. Create a symbolic link to enable the NGINX configuration:

   ```bash
   sudo ln -s /etc/nginx/sites-available/mtproxy /etc/nginx/sites-enabled/
   ```

6. Test the NGINX configuration for syntax errors:

   ```bash
   sudo nginx -t
   ```

   If there are no errors, reload NGINX to apply the changes:

   ```bash
   sudo systemctl reload nginx
   ```

7. Verify that NGINX is proxying requests to your MTProxy server by accessing `https://yourdomain.com` in a web browser or using a tool like `curl`.

   ```bash
   curl -x https://yourdomain.com https://api.telegram.org/bot<your_bot_token>/getMe
   ```

   Replace `<your_bot_token>` with your actual Telegram bot token.

Now, NGINX should be configured to proxy requests to your MTProxy server, allowing access to your MTProxy server via `https://yourdomain.com`.

**Reply by GadskyPapa** [reply]
Hello, I'm too lazy to prove it, but it shouldn't work. Just Google the phrase "nginx stream module proxy mtproto"

**Comment by tooker0** [comment]
Can I do the same with Apache? I tried almost everything but it doesn't work

**Comment by GadskyPapa** [comment]
The configuration is neither primitive nor really complicated - just use AI to help you. It could be a good starting point: https://share.google/aimode/2BwEOmkhFE0YqAsGT
